### PR TITLE
Fixes powerlines glyph rendering and transparency in webgl renderer

### LINKED ISSUE #2252: Webgl: Support allowTransparency option
Webgl addon added in https://github.com/xtermjs/xterm.js/pull/1790

I'm not sure what the state of this is currently.

### LINKED ISSUE #3342: Wide nerd font characters are cut to single width
Powerlines chars were trimmed to only fill a single cell in https://github.com/xtermjs/xterm.js/pull/3279, we need to look at the character codes in nerd fonts and make sure wide ones are trimmed to 2 cells.

Actual:

image

Expected (Terminal.app):

image

VS Code issue: https://github.com/microsoft/vscode/issues/123629

### PR DESCRIPTION
**Pull request by jeanp413**
Fixes #3342 and fixes #2252

Started playing around with the webgl renderer and seems I fixed these two issues, I've done some testing locally and it looks good  so far, would be great if someone could test this for some edge cases.

This still needs some cleanup as some code is commented out

### CODE REVIEW

File: addons/xterm-addon-webgl/src/GlyphRenderer.ts [hunk 1]
```diff
@@ -152,12 +152,14 @@ export class GlyphRenderer {
...
     gl.bindTexture(gl.TEXTURE_2D, this._atlasTexture);
     gl.texImage2D(gl.TEXTURE_2D, 0, gl.RGBA, 1, 1, 0, gl.RGBA, gl.UNSIGNED_BYTE, new Uint8Array([0, 0, 255, 255]));
+    gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MIN_FILTER, gl.LINEAR);
     gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_WRAP_S, gl.CLAMP_TO_EDGE);
     gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_WRAP_T, gl.CLAMP_TO_EDGE);
 
     // Allow drawing of transparent texture
     gl.enable(gl.BLEND);
-    gl.blendFunc(gl.SRC_ALPHA, gl.ONE_MINUS_SRC_ALPHA);
+    gl.blendEquationSeparate(gl.FUNC_ADD, (gl as any).MAX);
```
Comment (jeanp413) [inline addons/xterm-addon-webgl/src/GlyphRenderer.ts:161]: Seems the typings are incomplete as this should be available in a webgl2 context
